storage/country.cpp

```diff
@@ -18,119 +18,156 @@ namespace storage
 // all mwms in a subtree.
 using TMwmSubtreeAttrs = pair<uint32_t, size_t>;
 
-template <class ToDo>
-TMwmSubtreeAttrs LoadGroupSingleMwmsImpl(int depth, json_t * group, TCountryId const & parent, ToDo & toDo)
+namespace
 {
-  uint32_t mwmCounter = 0;
-  size_t mwmSize = 0;
-  size_t const groupListSize = json_array_size(group);
-  for (size_t i = 0; i < groupListSize; ++i)
-  {
-    json_t * j = json_array_get(group, i);
+class StoreSingleMwmInterface
+{
+public:
+  virtual Country * InsertToCountryTree(TCountryId const & id, uint32_t mapSize, int depth,
+                                        TCountryId const & parent) = 0;
+  virtual void InsertOldMwmMapping(TCountryId const & newId, TCountryId const & oldId) = 0;
```

Comment:
I did not quite catch. Are you talking about something like that:
```
class StoreSingleMwmInterface
{
public:
  virtual ~StoreSingleMwmInterface() {}
  virtual Country * InsertToCountryTree(TCountryId const & id, uint32_t mapSize, int depth,
                                        TCountryId const & parent) {return nullptr;}
  virtual void InsertOldMwmMapping(TCountryId const & newId, TCountryId const & oldId) {}
  virtual void InsertAffiliation(TCountryId const & countryId, string const & affilation) {}
  virtual TMappingOldMwm GetMapping() const {return ...}
};
```
?

If so, what is the advantage?
If not, what do you offer?